x/evm/keeper/abci.go

```diff
@@ -30,6 +33,12 @@ func (k *Keeper) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
 
 	// Gas costs are handled within msg handler so costs should be ignored
 	ctx.SetGasMeter(sdk.NewInfiniteGasMeter())
+	// load the preEIP effective height once
```

Comment:
It's better to explain why load once is fine, otherwise it is confusing in my opinion.